src/schema/rules/dataset_metadata.yaml

```diff
@@ -0,0 +1,67 @@
+---
+dataset_description:
+  selectors:
+  - path == "/dataset_description.json"
+  fields:
+    Name: REQUIRED
+    BIDSVersion: REQUIRED
+    DatasetType: RECOMMENDED
+    License: RECOMMENDED
+    Authors:
+      level: RECOMMENDED
+      issue:
+        code: NO_AUTHORS
+        message: |
+          The Authors field of dataset_description.json should contain an array of fields -
+          with one author per field. This was triggered because there are no authors, which
+          will make DOI registration from dataset metadata impossible.
+    Acknowledgements: OPTIONAL
+    HowToAcknowledge: OPTIONAL
+    Funding: OPTIONAL
+    EthicsApprovals: OPTIONAL
+    ReferencesAndLinks: OPTIONAL
+    DatasetDOI: OPTIONAL
+    GeneratedBy: RECOMMENDED
+    GeneratedBy[].Name: REQUIRED
+    GeneratedBy[].Version: RECOMMENDED
+    GeneratedBy[].Description: OPTIONAL
+    GeneratedBy[].CodeURL: OPTIONAL
+    GeneratedBy[].Container: OPTIONAL
+    GeneratedBy[].Container{}.Type: RECOMMENDED
+    GeneratedBy[].Container{}.Tag: RECOMMENDED
+    GeneratedBy[].Container{}.URI: RECOMMENDED
```

Comment:
Those dot, brackets and curly braces will MATLAB cry: the only special characters allowed in keys by matlab are `_` 

Related to 
https://github.com/bids-standard/bids-specification/pull/1082

Reply:
We need some way to mark the requirement levels of sub objects.

Reply:
See also: https://github.com/bids-standard/bids-specification/issues/841

Reply:
#841 is really just an artifact at this point. Once someone (probably me) writes a function to render subobjects in tables, we can get rid of the `Genetics.*` fields.